### PR TITLE
realtime username validation to check if it's available

### PR DESCRIPTION
**Pull request by CleverFool77**
Fixes https://github.com/publiclab/plots2/issues/5429 (<=== Add issue number here)

**Modal**
ezgif-2-0b74a620e952

**Signup**
signup

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

Thanks!

### CODE REVIEW

**Comment by SidharthBansal** [comment]
Please resolve code climate suggestions. 
Thanks for the work done here at PL

**Comment by CleverFool77** [comment]
Hi @SidharthBansal I tried refactoring it, But We can't change the issue of similarity that codeclimate is showing. As both of them are different cases.
Or do you have some suggestion, maybe you can help with it ?

**Comment by SidharthBansal** [comment]
No problem. We can ignore code climate warnings.
Can you please illustrate the behaviour of the signup page too?

**Comment by SidharthBansal** [comment]
No problem. We can ignore code climate warnings.
Can you please illustrate the behaviour of the signup page too?

**Comment by CleverFool77** [comment]
Hi @SidharthBansal This PR is working for signup page too . I'll update the gif.
Thanks !!

**Comment by CleverFool77** [comment]
Hi I've updated the PR and gif.
please review.
thanks !!

File: app/views/users/_create_form.html.erb [hunk 1]
```diff
@@ -25,7 +25,8 @@
         <div class="form-group col-sm-12" id="username_div">
           <label for="username"><%= t('user_sessions.new.username') %></label>
           <%= f.text_field :username, { tabindex: 1, placeholder: "Username", class: 'form-control', id: 'username-signup' } %>
-        </div>
+          <label class="username-check"></label>
+          </div>
```
Comment (gauravano) [inline app/views/users/_create_form.html.erb]:
```suggestion
        </div>
```

**Comment by CleverFool77** [comment]
@gauravano I've updated the PR as requested :+1:

**Comment by gauravano** [comment]
Merged :tada: :100: :+1: !

@jywarren check this out!!

**Pull request merged**